src/Features/VisualBasic/Portable/GoToDefinition/VisualBasicGoToDefinitionSymbolService.vb

```diff
@@ -20,5 +21,141 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.GoToDefinition
         Protected Overrides Function FindRelatedExplicitlyDeclaredSymbol(symbol As ISymbol, compilation As Compilation) As ISymbol
             Return symbol.FindRelatedExplicitlyDeclaredSymbol(compilation)
         End Function
+
+        Protected Overrides Function GetTargetPositionIfControlFlow(token As SyntaxToken) As Integer?
+            Dim node = token.GetRequiredParent()
+
+            If token.IsKind(SyntaxKind.ReturnKeyword, SyntaxKind.YieldKeyword) Then
+                Return FindContainingReturnableConstruct(node).GetFirstToken().Span.Start
+            End If
+
+            Dim continueTarget = TryGetContinueTarget(node)
+            If continueTarget IsNot Nothing Then
+                Return continueTarget.GetFirstToken().Span.Start
+            End If
+
+            Dim exitTarget = TryGetExitTarget(node)
+            If exitTarget IsNot Nothing Then
+                Select Case node.Kind()
+                    Case SyntaxKind.ExitSubStatement
+                    Case SyntaxKind.ExitFunctionStatement
+                    Case SyntaxKind.ExitPropertyStatement
+                        Return exitTarget.GetFirstToken().Span.Start
+                End Select
+
+                Return exitTarget.GetLastToken().Span.End
```

Comment:
can you doc the distinction hre?  why do some jump to the first token, and some to the last?